server/questions.js

```diff
@@ -196,7 +202,35 @@ exports.checkAnswer = function(questionId, user, answer, callback) {
 			return callback('error', null);
 		}
 
-		var value = (answer === question.answer);
+		var value = null;
+
+		if (question.type === common.questionTypes.MATCHING.value && answer) {
```

Comment:
I dont like this. Its a huge addition to the checkanswer function for only one new type. We need a better way to check the answers based on type. I will let this one in, but I will refactor all of this later